e2e/tests/test_workload_scripts.py

```
@@ -0,0 +1,113 @@
+        workloads.source_command("test-image", gpu=gpu),
+    ):
+        assert workloads.SNAPSHOT_COMPLETE not in script
+        assert workloads.SOURCE_READY in script
+
+
+@pytest.mark.workload
+def test_snapshotjob_exit_template_never_signals_ready() -> None:
+    # The exit templates drive the died-before-capture failure classes; they
+    # must terminate without touching the quiesce protocol at all.
+    for exit_code in (0, 1):
+        proc = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)